qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -64,6 +69,96 @@ protected boolean preserveReposUponCompletion() {
         return true;
     }
 
+    public void testOldIndexBwc() throws Exception {
```

Comment:
Maybe `testSearch` instead?

I think it'd be nice to move some of the random document generation from `testRandomDocumentsAndSnapshot` below into this and just use `testRandomDocumentsAndSnapshot` for testing the translog stuff.